### PR TITLE
feat(plugin): add squad plugin install <repo> command

### LINKED ISSUE #972: feat: add `squad plugin install <owner/repo>` CLI command for flat extensions
## Problem

The plugin marketplace docs ([Building Extensions](https://bradygaster.github.io/squad/docs/guide/building-extensions/), [Extensibility Guide](https://bradygaster.github.io/squad/docs/guide/extensibility/)) describe installing extensions with:

```
squad plugin install github/my-org/my-extension
```

But the CLI only supports `squad plugin marketplace add|remove|list|browse` — there is no `install` subcommand. This makes it impossible to install an extension into a repo via the CLI.

## Current workaround

Manually clone the extension repo and copy `.md` files into the target repo's `.squad/skills/`, `.squad/ceremonies/`, and `.squad/directives/` directories.

## Expected behavior

A command like:

```bash
squad plugin install <owner/repo>
```

Should:

1. Clone or fetch the extension repo
2. Detect the extension structure (skills/, ceremonies/, directives/ at root)
3. Copy the `.md` files into the target repo's `.squad/` subdirectories
4. Log what was installed

## Additional context

- Extensions like [github-squad-workflows](https://github.com/sabbour/github-squad-workflows) follow the flat extension structure (`skills/`, `ceremonies/`, `directives/` at root) as documented in the [Building Extensions](https://bradygaster.github.io/squad/docs/guide/building-extensions/) guide.
- The `browse` command treats top-level directories as individual plugins, which does not work well for flat extensions — it shows `skills`, `ceremonies`, `directives` as plugin names instead of meaningful entries.
- Squad version: 0.9.1-build.5

## Proposal

1. Add `squad plugin install <owner/repo>` as a CLI command
2. Support flat extension repos (skills/ceremonies/directives at root) in addition to marketplace-style repos (plugin subdirectories)
3. Optionally track installed extensions in `.squad/plugins/installed.json` for upgrade/remove support

### PR DESCRIPTION
**Pull request by sabbour**
## Summary

Adds `squad plugin install <repo>` CLI command for installing flat extensions from GitHub repos.

### What it does

1. **Parses repo references** — accepts both `owner/repo` and `github/owner/repo` formats
2. **Clones the extension repo** (shallow, depth 1) into a temp directory
3. **Detects extension structure** — looks for `skills/`, `ceremonies/`, `directives/` at repo root
4. **Copies `.md` files** into the project's `.squad/` directory, creating directories as needed
5. **Tracks installed plugins** in `.squad/plugins/installed.json` for future upgrade/remove support
6. **Cleans up** the temp clone directory in a `finally` block

### Files changed

- `packages/squad-cli/src/cli/commands/plugin.ts` — added install subcommand, helpers, types
- `packages/squad-cli/package.json` — added `./commands/plugin` export
- `test/cli/plugin-install.test.ts` — 17 tests (parsing, file collection, install flow, reinstall, edge cases)
- `.changeset/plugin-install-command.md` — minor changeset

### Testing

All 17 new tests pass. Existing marketplace tests (157 tests) and CLI wiring tests (32 tests) unaffected.

➜  kickstart git:(squad/145-fix-system-prompts-a2ui) ✗ squad plugin install sabbour/github-squad-workflows
Cloning sabbour/github-squad-workflows…
  📄 skills/pr-workflow.md → .squad/skills/pr-workflow.md
  📄 skills/release-process.md → .squad/skills/release-process.md
  📄 skills/testing-strategy.md → .squad/skills/testing-strategy.md
  📄 ceremonies/design-review.md → .squad/ceremonies/design-review.md
  📄 ceremonies/retrospective.md → .squad/ceremonies/retrospective.md
  📄 ceremonies/sprint-lifecycle.md → .squad/ceremonies/sprint-lifecycle.md
  📄 ceremonies/sprint-planning.md → .squad/ceremonies/sprint-planning.md
  📄 directives/workflow-conventions.md → .squad/directives/workflow-conventions.md
✓ Installed github-squad-workflows — 8 file(s) copied

Closes bradygaster/squad#972

🤖 Created by [sabbour-squad-backend](https://github.com/apps/sabbour-squad-backend)

### CODE REVIEW

File: packages/squad-cli/src/cli/commands/plugin.ts [hunk 1]
```diff
@@ -26,14 +26,170 @@ export interface MarketplacesRegistry {
...
+      for (const file of mdFiles) {
+        const srcPath = join(srcDir, file);
+        const destPath = join(destDir, file);
+        copyFileSync(srcPath, destPath);
+        installedFiles.push({
+          source: `${dirName}/${file}`,
+          dest: destPath,
+        });
+        info(`  📄 ${dirName}/${file} → .squad/${dirName}/${file}`);
+      }
```
[the remark this answers is not in the document]
Comment (sabbour) [reply packages/squad-cli/src/cli/commands/plugin.ts:129]: @copilot apply changes based on [this feedback](https://github.com/bradygaster/squad/pull/973#discussion_r3077370334)

**Comment by sabbour** [comment]
Closing as needing further discussion around extensions and plugins.
